[new-product] add Kong Gateway (#3515)
captn3m0/endoflife.date: products/kong.md

---
title: Kong
category: server-app

> tags: gateway

Reviewer: ```suggestion
tags: api-gateway
```

`gateway` is too generic.

permalink: /kong
releasePolicyLink: https://docs.konghq.com/gateway/latest/support-policy/

> changelogTemplate: https://docs.konghq.com/gateway/changelog/#{{'__RELEASE_CYCLE__'|split:'.'|pop|join:'-'}}

Reviewer: ```suggestion
changelogTemplate: "https://docs.konghq.com/gateway/changelog/#{{'__LATEST__'|replace:'.',''}}0"
```

This produces links such as `https://docs.konghq.com/gateway/changelog/#$x`, where `$x` is the major version number. Something like that should work better (assuming community version are always x.y.z.0 versions).

Author: good point! There are versions that does not end in 0 in the changelog([example](https://docs.konghq.com/gateway/changelog/#3223)) but for every cycle the community version does seem to end in 0 so that's a good one, I will change it

EDIT: just came to my mind that this x.y.z.a are the enterprise patches notation, and that every community version simply is the .0.0 enterprise patch

releaseDateColumn: true

> activeSupportColumn: false

Reviewer: ```suggestion
```

Not needed, this is the default value.

eolColumn: Support

auto:
  - git: https://github.com/Kong/kong.git

identifiers:
  - purl: pkg:github/Kong/kong

releases:
  - releaseCycle: "3.4"
    releaseDate: 2023-08-09
    eol: 2026-08-09
    lts: true
    latest: "3.4.0"
    latestReleaseDate: 2023-08-09

  - releaseCycle: "3.3"
    releaseDate: 2023-05-18
    eol: 2024-05-18

>     lts: false

Reviewer: ```suggestion
```

Not needed as `false` is already the default version.

    latest: "3.3.1"
    latestReleaseDate: 2023-07-11

  - releaseCycle: "3.2"
    releaseDate: 2023-02-20
    eol: 2024-02-20

>     lts: false

Reviewer: ```suggestion
```

Not needed as `false` is already the default version.

    latest: "3.2.2"
    latestReleaseDate: 2023-03-16

  - releaseCycle: "3.1"
    releaseDate: 2022-06-12
    eol: 2023-06-12

>     lts: false

Reviewer: ```suggestion
```

Not needed as `false` is already the default version.

    latest: "3.1.1"
    latestReleaseDate: 2022-12-09

  - releaseCycle: "3.0"
    releaseDate: 2022-09-12
    eol: 2023-09-12

>     lts: false

Reviewer: ```suggestion
```

Not needed as `false` is already the default version.

    latest: "3.0.2"
    latestReleaseDate: 2022-12-09

  - releaseCycle: "2.8"
    releaseDate: 2022-03-01

>     eol: 2023-03-01

Reviewer: ```suggestion
    eol: 2025-03-01
```

This is an LTS (see https://docs.konghq.com/gateway/latest/support-policy/).

    lts: true
    latest: "2.8.3"
    latestReleaseDate: 2022-11-03
---

> Kong or Kong API Gateway is a cloud-native, platform-agnostic, scalable API Gateway distinguished for its high performance and extensibility via plugins.
> By providing functionality for proxying, routing, load balancing, health checking, authentication (and more), Kong serves as the central layer for orchestrating microservices or conventional API traffic with ease.

> [Kong Support Policy](https://docs.konghq.com/gateway/latest/support-policy/) declares
> that each minor version is supported for 1 year after release.
> 
> A minor version can be designated as a LTS version, which has support for the next 3 years

Reviewer: ```suggestion
Kong follows [Semantic Versioning](https://semver.org/). Minor versions are released every 10 weeks and supported for 1 year with bug and security fixes. Some minor versions are designated LTS and supported for 3 years.
```

WDYT of this ?

I also remove the support policy link because it is already provided with the `releasePolicyLink` (see https://deploy-preview-3515--endoflife-date.netlify.app/kong).

Author: Well noted!

> A minor version can be designated as a LTS version, which has support for the next 3 years

Reviewer: Maybe something can be said about Kong Enterprise ?

Author: Sure! I provided a quickly explanation on Enterprise patches based on the Kong Community as per the Release Policy doc

Reviewer: I have re-read https://docs.konghq.com/gateway/latest/support-policy/ and think we may have made a mistake: this policy only applies to enterprise, not community (as said in the title of the page).

Looking at the latest community release dates it looks like they _usually_ only support the latest minor release. WDYT ? Could you contact them to clarify ?

Author: Yeah I got confused at first also, It seems to me that they use the Community Versions to patch Enterprise Versions. So the policy to the Enterprise Patches applies to Community Versions to some extent.

Example: Take version 3.0.0 which was released Sep 12, 2022, by the policy as we understood the full support ends on Sept 12, 2023. The last version updated on 3.0 was 3.0.2 which was released Dec 9, 2022 (a little further than the official policy, maybe because of the Sunset Support), and between those dates they released 2.8.3 which is the LTS going until March 2025. 

So that being said to me it seems that they are on par with their policy on Community despite the title of the page saying that is for Enterprise, also because they mention the Community Versions in one of their paragraphs. But it is indeed confusing, this is not even the third time that this happens understading diffs between Enterprise and Community for Kong :laughing:. So to address this I posted this [comment on their forum](https://discuss.konghq.com/t/question-about-the-support-policy-on-kong-community/11891?u=alexandre_ferreira) to guarantee that our understanding is correct (couldn't access their slack :cry:), let's wait a few to see their response, if not we assume that we understood correctly based on the release dates on their releases in github

Author: @marcwrobel we've got no response on their forum; WDYT on merging this based on our understanding and hopefully the community will catch on any gaps on our development?

Reviewer: @aleferreiranogueira I updated the page to:

- rename kong to kong gateway (Kong Inc. has now multiple products named Kong <product>).
- make it explicit that the page is about the community version
- say explicitly what I think about the release / support policy
- add an extended support column to document the enterprise support

WDYT of this updated page: https://deploy-preview-3515--endoflife-date.netlify.app/kong-gateway ?

Author: @marcwrobel you sir indeed are a gentleman of good taste, LGTM!

![](https://media.giphy.com/media/kwcRp24Wz4lZm/giphy.gif)
